tests/core/contracts/test_contract_call_interface.py

```diff
@@ -314,3 +324,44 @@ def test_call_undeployed_contract(undeployed_math_contract):
     with pytest.raises(BadFunctionCallOutput) as exception_info:
         undeployed_math_contract.functions.return13().call()
     assert expected_undeployed_call_error_message in str(exception_info.value)
+
+
+def test_throws_error_if_block_out_of_range(web3, math_contract):
+    web3.providers[0].make_request(method='evm_mine', params=[20])
+    with pytest.raises(BlockNumberOutofRange):
+        math_contract.functions.counter().call(block_identifier=50)
+
+    with pytest.raises(BlockNumberOutofRange):
+        math_contract.functions.counter().call(block_identifier=-50)
+
+
+@patch('web3.eth.Eth.call', return_value='0x' + '0' * 64)
```

Comment:
I'd like to try and remove the mocks.

Since we're using the `Math` contract which has the `increment` method, it should be pretty trivial for us to execute a few transactions incrementing the counter and then execute the calls against those block numbers checking that the `counter` value is what we expect.

Reply:
Cool yeah 100% agree, I built it that way originally until I realized `eth-tester` didn't keep past blocks. Let me know when that's merged and I'll update the code here.

Reply:
@pipermerriam I don't want this to fall through the cracks. Did the eth-tester updates ever get merged to be able to finish this? If not I'd be happy to take a look at tackling that issue.

Reply:
There have been a lot of updates merged and released into eth-tester.  I'd suggest rebasing your PR and seeing if the issue you ran into is fixed.

Otherwise, I'm not sure I follow your statement:

> until I realized `eth-tester` didn't keep past blocks.

I don't understand what you're referring to here.  Can you expand?

Reply:
Ok I can rebase. The problem I was having with the `eth-tester` backends was that they only allowed me to query the state of the last block, so I couldn't increment the counter from one block to the next and then query the previous block's state, which caused me to have to do the Mock's in my tests on `web3.eth.call` (thus making the implicit assumption that the `block_identifier` parameter works within that method) instead of directly querying the previous state. Does that make sense?

Reply:
Can you try rebasing and running again? It looks to me like eth-tester supports it: https://github.com/ethereum/eth-tester/blame/master/eth_tester/main.py#L406

Although I'm a little confused about:
> The tests for this feature feel partially incomplete because the eth-tester package does not allow one to query past contract state (outlined here ethereum/eth-tester#47)

because it seems like that feature has been there a while. Anyway, once this is rebased, we can take another look at where it's failing.